addons/portal/models/mail_message.py

```diff
@@ -8,6 +8,28 @@
 class MailMessage(models.Model):
     _inherit = 'mail.message'
 
+    def _compute_is_current_user_or_guest_author(self):
+        super()._compute_is_current_user_or_guest_author()
+        portal_data = self.env.context.get("portal_data", {})
+        portal_partner = portal_data.get("portal_partner")
+        portal_thread = portal_data.get("portal_thread")
+        if (
+            not portal_partner
+            or not portal_thread
+            or not isinstance(portal_partner, self.pool["res.partner"])
+            or not isinstance(portal_thread, self.pool["mail.thread"])
+        ):
+            return
```

Comment:
This should not be a quick return here, compute needs to assign value on all records